hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -231,6 +254,36 @@ private static GenericRecord rewrite(GenericRecord record, LinkedHashSet<Field>
     return allFields;
   }
 
+  /*
+   * Given a avro record with a given schema, rewrites it into the new schema while setting fields only from the old
+   * schema.
+   *
+   * NOTE: This function is only suitable if newSchema has fields with the same position as record's schema.
+   */
+  public static GenericRecord rewriteHoodieRecord(GenericRecord record, Schema newSchema) {
+    return rewriteHoodieRecord(record, record.getSchema(), newSchema);
```

Comment:
Please use getCombinedFieldsToWrite() function here, that will help in handling schema evolution cases also.